packages/state-manager/src/sagas/files/checkForMissingFiles/checkForMissingFiles.saga.ts

```diff
@@ -20,10 +24,20 @@ export function* checkForMissingFilesSaga(
 
   const channels = yield* select(publicChannelsSelectors.publicChannels)
 
+  const fileStatuses = yield* select(filesSelectors.downloadStatuses)
+
   for (const channel of channels) {
     const missingFiles = yield* select(missingChannelFiles(channel.address))
     if (missingFiles.length > 0) {
       for (const file of missingFiles) {
+        if (file.size > AUTODOWNLOAD_SIZE_LIMIT || fileStatuses[file.message.id].downloadState === DownloadState.Canceled) return
```

Comment:
Ideally it'd be two separate conditions, like that:

` if (file.size > AUTODOWNLOAD_SIZE_LIMIT) return`
` if (fileStatuses[file.message.id].downloadState === DownloadState.Canceled) return`

It's way better readable and easier to maintain